### PR TITLE
LCH-4746: Create BackupRestore and BackupDelete Commands

### PR DESCRIPTION
**Pull request by abarriosr**

### CODE REVIEW

File: src/Command/Backups/BackupDelete.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getExpectedPlatformOptions(): array {
+    return ['source' => PlatformCommandInterface::ANY_PLATFORM];
+  }
+
+  /**
+   * BackupCreate constructor.
```
Comment (amangrover90) [inline src/Command/Backups/BackupDelete.php:38]: Comment change for constructor name.

File: src/Command/Backups/BackupRestore.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getExpectedPlatformOptions(): array {
+    return ['source' => PlatformCommandInterface::ANY_PLATFORM];
+  }
+
+  /**
+   * BackupCreate constructor.
```
Comment (amangrover90) [inline src/Command/Backups/BackupRestore.php:39]: Comment change for constructor name.

**Comment by ankitsingh0188** [comment]
There's a minor change which we needs to be done in AcquiaCloudBackupRestore.php and AcquiaCloudBackupDelete.php

line no. 96
`$configs = array_map(function (Config $config) {
      return $config->get('name');
    }, $backups);`

After this we need to check whether $configs is exists or not, if not then we need to throw some error.

File: src/Command/Backups/BackupDelete.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  public function __construct(EventDispatcherInterface $event_dispatcher, string $name = NULL) {
+    parent::__construct($name);
+    $this->dispatcher = $event_dispatcher;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    $this->setDescription('Deletes backups bundles of Content Hub Service snapshots and database site backups.');
```
Comment (amangrover90) [inline src/Command/Backups/BackupDelete.php:54]: Can we change it to **Deletes a backup bundle of Acquia Content Hub Service snapshot and database site backups.** similar to what we have for BackupCreate or this is fine as well.

File: src/Command/Backups/BackupRestore.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  public function __construct(EventDispatcherInterface $event_dispatcher, string $name = NULL) {
+    parent::__construct($name);
+    $this->dispatcher = $event_dispatcher;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    $this->setDescription('Restores a backup bundle of Content Hub Service snapshots and database site backups created previously.');
```
Comment (amangrover90) [inline src/Command/Backups/BackupRestore.php:55]: Can we change it to **Restores a backup bundle of Acquia Content Hub Service snapshot and database site backups.** similar to what we have for BackupCreate or this is fine as well.

**Comment by ankitsingh0188** [comment]
In _commoncli.services.yml_ we can set `public: true` TO `public: false` for all the hidden commands

List of hidden commands:

1. AcquiaCloudBackupCreate
2. AcquiaCloudBackupCreateMultiSite
3. AcquiaCloudBackupDelete
4. AcquiaCloudBackupDeleteMultisite
5. AcquiaCloudBackupRestore
6. AcsfBackupCreate
7. AcsfBackupDelete
8. AcsfBackupRestore
9. ContentHubAuditChServices
10. ContentHubAuditDepcalc
11. ContentHubAuditSslCertificate
12. ContentHubAuditStreamWrappers
13. ContentHubAuditTmpFiles
14. ContentHubClientCompare
15. ContentHubDiff
16. ContentHubInterestListCheck
17. ContentHubModuleVersion
18. ContentHubQueue
19. ContentHubVerifyWebhooksDefaultFilters

**Note:** I think AcquiaCloudBackupRestoreMultisite command must be hidden.

**Review by ankitsingh0188 — APPROVED** [review]
Looks good to me!

**Review by amangrover90 — APPROVED** [review]

**Pull request merged**